### PR TITLE
[API] Return changeSet and validationMessages on dryrun response(s) t…

### PR DESCRIPTION
**Pull request by charlesg3**
…o create/update cluster.

*Description of changes:*
- Add `validationMessages` to the response when creating a cluster if `dryrun` flag is set
- Add `validationMessages` and `changeSet` to the response when updating a cluster if `dryrun` flag is set
- Update unit tests to reflect above change
- Update integration tests to check for appropriate responses during create / update when doing dryrun

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: cli/tests/pcluster/api/controllers/test_cluster_operations_controller.py [hunk 1]
```diff
@@ -1207,10 +1207,22 @@ def test_successful_update_request(
     def test_dryrun(self, mocker, client):
```
Comment (demartinofra) [inline cli/tests/pcluster/api/controllers/test_cluster_operations_controller.py:1207]: nit: shall we also test the suppressed validation messages? Aren't we testing dryrun for create?

**Review by demartinofra — COMMENTED** [review]
LGTM. Probably need to review integ tests + generate client code

File: cli/tests/pcluster/api/controllers/test_cluster_operations_controller.py (hunk 1 above)
Comment (charlesg3) [reply cli/tests/pcluster/api/controllers/test_cluster_operations_controller.py]: https://github.com/aws/aws-parallelcluster/pull/3122/commits/61fdaeae8c93895f9f0d9949604ac54ecb292d64

**Comment by charlesg3** [comment]
> LGTM. Probably need to review integ tests + generate client code
Generate client code: https://github.com/aws/aws-parallelcluster/pull/3122/commits/a34fe87261a2ad9fbb128859401a1e526b4a8233

**Comment by charlesg3** [comment]
> LGTM. Probably need to review integ tests + generate client code

Updated integration tests: https://github.com/aws/aws-parallelcluster/pull/3122/commits/fe8b2b191e8549b116c7de1101c10b384f3d11ec
- Refactored the code to be more data-driven -- describes the various combinations of suppression / warning vs error / output / dryrun (and it is validated for both create and update operations.)

File: tests/integration-tests/tests/cli_commands/test_cli_commands.py [hunk 2]
```diff
@@ -357,61 +303,59 @@ def _test_pcluster_export_cluster_logs(s3_bucket_factory, cluster):
...
+    ip_str = instance["privateIpAddress"].replace(".", "-")
+    return "ip-{}.{}.{}".format(ip_str, instance["instanceId"], stream_name)
+
+
 def _test_pcluster_list_cluster_log_streams(cluster):
     """Test pcluster list-cluster-logs functionality and return cfn-init log stream name."""
     logging.info("Testing that pcluster list-cluster-log-streams is working as expected")
     list_streams_result = cluster.list_log_streams()
+    cluster_info = cluster.describe_cluster()
     streams = list_streams_result["items"]
```
Comment (demartinofra) [inline tests/integration-tests/tests/cli_commands/test_cli_commands.py:339]: probably this needs to be updated

File: tests/integration-tests/utils.py [hunk 3]
```diff
@@ -208,12 +204,9 @@ def get_cluster_nodes_instance_ids(stack_name, region, instance_types=None, node
...
             filter_by_queue_name=queue_name,
         )
-        instance_ids = []
-        for instance in instances:
-            instance_ids.append(instance.get("InstanceId"))
-        return instance_ids
+        return [instance["InstanceId"] for instance in instances]
     except Exception as e:
-        logging.error("Failed retrieving instance ids with exception: {}".format(e))
+        logging.error("Failed retrieving instance ids with exception: %", e)
```
Comment (demartinofra) [inline tests/integration-tests/utils.py:209]: %s?

**Review by demartinofra — APPROVED** [review]

**Pull request merged**